### PR TITLE
update `write-flows` docs

### PR DESCRIPTION
**Pull request by zzstoatzz**

### CODE REVIEW

File: docs/3.0rc/develop/write-flows.mdx [hunk 1]
```diff
@@ -3,41 +3,42 @@ title: Write and run flows
 description: Learn the basics of defining and running flows.
 ---
 
-Flows are the most central Prefect objects.
```
Comment (zzstoatzz) [inline docs/3.0rc/develop/write-flows.mdx:6]: i dont think this means anything useful to readers

File: docs/3.0rc/develop/write-flows.mdx [hunk 2]
```diff
@@ -3,41 +3,42 @@ title: Write and run flows
...
-Make any function a Prefect flow by adding the `@flow` decorator to it:
+Make any Python function a Prefect flow by adding the `@flow` decorator to it:
 
 ```python
 from prefect import flow
 
 
 @flow
-def my_flow():
-    return
```
Comment (zzstoatzz) [inline docs/3.0rc/develop/write-flows.mdx:20]: i think all examples (within reason) should be directly copy / paste-able and runnable

File: docs/3.0rc/develop/write-flows.mdx [hunk 3]
```diff
@@ -99,22 +100,6 @@ Prefect automatically performs type conversion of inputs using any provided type
...
-
-
-class Model(BaseModel):
-    a: int
-    b: float
-    c: str
-
-
-@flow
-def model_validator(model: Model):
```
Comment (zzstoatzz) [inline docs/3.0rc/develop/write-flows.mdx:114]: imo we should not use names for flows that shadow popular other well-established names

File: docs/3.0rc/develop/write-flows.mdx [hunk 4]
```diff
@@ -144,6 +129,36 @@ When you run this flow, you'll see the following output:
...
+
+@flow
+def flow_that_validates_parameters(model: Model): ...
+
+if __name__ == "__main__":
+    try:
+        flow_that_validates_parameters(
+            model={"a": "WRONG", "b": "fine"}
+        )
+    except Exception as e:
```
Comment (cicdw) [inline docs/3.0rc/develop/write-flows.mdx:153]: I don't think I understand the `try/except/print` here - wouldn't the error get raised and displayed normal-style without this?

**Review by cicdw — APPROVED** [review]
nice!! one suggestion on a code example, otherwise LGTM

File: docs/3.0rc/develop/write-flows.mdx (hunk 4 above)
Comment (zzstoatzz) [reply docs/3.0rc/develop/write-flows.mdx]: :doh: yeah you're totally right, just a lapse on my end

updated in [ba904c1](https://github.com/PrefectHQ/prefect/pull/15146/commits/ba904c168a824e445013a011067f080ed60e39fd)

**Pull request merged**